pt/lessons/libraries/benchee.md

```diff
@@ -0,0 +1,333 @@
+
+```elixir
+Benchee.run(%{"example function" => fn -> "hi!" end}, [
+  warmup: 4,
+  time: 10,
+  inputs: nil,
+  parallel: 1,
+  formatters: [&Benchee.Formatters.Console.output/1],
+  print: [ 
+    benchmarking: true,
+    configuration: true,
+    fast_warning: true
+  ],
+  console: [
+    comparison: true,
+    unit_scaling: :best
+  ]
+])
+```
+
+As opções disponíveis são as seguintes (também documentadas em [hexdocs](https://hexdocs.pm/benchee/Benchee.Configuration.html#init/1)).
+
+* **warmup** - o tempo em segundos para o qual um cenário de _benchmarking_ deve ser executado sem tempos de medição antes do início das medidas reais. Isso simula um sistema de funcionamento "quente". Padrão é 2.
+* **time** - o tempo em segundos por quanto tempo cada cenário de _benchmarking_ individual deve ser executado e medido. Padrão é 5.
+* **inputs** - um mapa com _strings_ que representam o nome de entrada como as chaves e a entrada real como valores. Padrão é `nil`. Vamos abordá-lo em detalhes na próxima seção.
+* **parallel** - o número de processos para usar no _benchmark_ de suas funções. Então, se você definir `parallel: 4`, serão gerados 4 processos que executam a mesma função para determinado `time`. Quando estes terminam, então 4 novos processos serão gerados para a próxima função. Isso lhe dá mais dados no mesmo tempo, mas também adiciona mais carga ao sistema interferindo nos resultados do _benchmark_. Isso pode ser útil para simular um sistema sobre carga o que algumas vezes é útil, mas deve ser usado com algum cuidado pois isso pode afetar os resultados de maneiras imprevisíveis. Padrão é 1 (o que significa nenhuma execução em paralelo). 
+* **formatters** - a list of formatter functions you'd like to run to output the benchmarking results of the suite when using `Benchee.run/2`. Funções precisam aceitar um argumento (que é a suite de _benchmarking_ para todos dados) e então usá-la para produzir a saída. Padrão é o formatador de console embutido chamando `Benchee.Formatters.Console.output/1`. Vamos abordar mais sobre isso em uma seção posterior.
+* **print** - um _map_ ou _keyword list_ com as seguintes opções como átomos para as chaves e valores de `true` ou `false`. Isso nos permite controlar se a saída identificada pelo átomo será impressa durante o processo padrão de _benchmarking_. Todas as opções são habilitadas por padrão (true). Opções são:
+  * **benchmarking** - imprime quando Benchee inicia o _bencharking_ de um novo _job_.
+  * **configuration** - um resumo de opções de _benchmarking_ configuradas incluindo o tempo de execução total estimado é impresso antes do _benchmarking_ iniciar.
+  * **fast_warning** - avisos são mostrados se funções são executadas muito rápidas, potencialmente levando a medidas imprecisas.
+* **console** - um _map_ ou _keyword list_ com as seguintes opções como átomos para as chaves e valores de variáveis. Os valores de variáveis são listados sobre cada opção:
```

Comment:
I would change "(...) e valores de variáveis." to "(...) e valores variáveis."

Likewise, "(...) são listados sobre cada opção:" to "são listados para cada opção:".